tests/test_trainer.py

```diff
@@ -249,8 +253,12 @@ def mock_save_function(filepath):
     # CASE K=0 (none)
     w = ModelCheckpoint(save_dir, save_top_k=0, verbose=1)
     w.save_function = mock_save_function
+    trainer = Trainer()
+    w.set_trainer(trainer)
     for i, loss in enumerate(losses):
-        w.on_epoch_end(i, logs={'val_loss': loss})
+        w._trainer.current_epoch = i
+        w._trainer.callback_metrics = {'val_loss': loss}
```

Comment:
add a comment that this is kind of hack to simulate training...

Reply:
Done